packages/sales-api-service/src/services/recurring-payments.service.js

```diff
@@ -126,3 +126,12 @@ export const processRPResult = async (transactionId, paymentId, createdDate) =>
 
   return { permission }
 }
+
+export const findNewestExistingRecurringPaymentInCrm = async agreementId => {
+  const matchingRecords = await executeQuery(findRecurringPaymentsByAgreementId(agreementId))
+  if (matchingRecords) {
+    return [...matchingRecords].sort((a, b) => a.endDate.localeCompare(b.endDate)).pop()
+  } else {
+    return false
```

Comment:
Doesn't look like you're testing for this - you test for returnedRecord to be `undefined` but not a boolean `false`